crates/forge_analyzer/src/checkers.rs

```diff
@@ -926,61 +927,60 @@ impl<'cx> Runner<'cx> for SecretChecker {
+                            let vuln =
+                                SecretVuln::new(interp.callstack(), interp.env(), interp.entry());
                             info!("Found a vuln!");
                             self.vulns.push(vuln);
                         }
                     }
-                    Operand::Var(var) => {
-                        if let Base::Var(varid) = var.base {
-                            if let Some(value) =
-                                interp.get_value(def, varid, Some(var.projections.clone()))
-                            {
-                                match value {
-                                    Value::Const(_) | Value::Phi(_) => {
-                                        let vuln = SecretVuln::new(
-                                            interp.callstack(),
-                                            interp.env(),
-                                            interp.entry(),
-                                        );
-                                        info!("Found a vuln!");
-                                        self.vulns.push(vuln);
-                                    }
-                                    _ => {}
-                                }
-                            } else if let Some(VarKind::GlobalRef(def)) =
-                                interp.body().vars.get(varid)
-                            {
-                                if let Some(value) = interp.value_manager.defid_to_value.get(def) {
-                                    println!("value [] {value:?}");
-                                    match value {
-                                        Value::Const(_) | Value::Phi(_) => {
-                                            let vuln = SecretVuln::new(
-                                                interp.callstack(),
-                                                interp.env(),
-                                                interp.entry(),
-                                            );
-                                            info!("Found a vuln!");
-                                            self.vulns.push(vuln);
-                                        }
-                                        _ => {}
-                                    }
-                                }
-                            }
+                }
+            }
+            Intrinsic::Fetch => {
+                if let Some(Operand::Var(Variable {
+                    base: Base::Var(varid),
+                    ..
+                })) = operands.unwrap_or_default().get(1)
+                {
+                    let varid_argument =
+                        if let Some(Value::Object(varid)) = interp.get_value(def, *varid, None) {
+                            varid
+                        } else {
+                            varid
+                        };
+                    let headers_proj = projvec_from_str("Headers");
```

Comment:
The lowercase/uppercase normalization only applies to the actual headers themselves, not the headers property.

Reply:
fixed. Thank you!